core/Pos/Binary/Class/Primitive.hs

```diff
@@ -126,100 +151,7 @@ class AsBinaryClass a where
-    UnsignedVarInt (len :: Int64) <- limitGet 20 get
-    if len <= lim
-        then isolate64Full len (getter len)
-        else fail $ formatToString
-                      ("getWithLengthLimited: data ("%int%" bytes) is "%
-                       "bigger than the limit ("%int%" bytes)")
-                      len lim
-
--- | Like 'putWithLength', but should only be used for things that take less
--- than @2^14@ bytes.
---
--- Uses 'TinyVarInt' for storing length, thus guaranteeing that it won't take
--- more than 2 bytes and won't be ambiguous.
-putSmallWithLength :: PokeWithSize a -> Poke a
-putSmallWithLength = pwsToPoke . putSmallWithLengthS
-
--- | Like @putSmallWithLength@ but returns PokeWithSize.
-putSmallWithLengthS :: PokeWithSize a -> PokeWithSize a
-putSmallWithLengthS a@(PokeWithSize len _) =
-    if len >= 2^(14::Int)
-        then error ("putSmallWithLength: length is " <> show len <>
-                    ", but maximum allowed is 16383 (2^14-1)")
-        else putS (TinyVarInt (fromIntegral len)) *> a
-
--- | Like 'getWithLength' but for 'putSmallWithLength'.
-getSmallWithLength :: (Word16 -> Peek a) -> Peek a
-getSmallWithLength getter = do
-    TinyVarInt len <- get
-    isolate64Full (fromIntegral len) (getter len)
-
-----------------------------------------------------------------------------
--- Other binary utils
-----------------------------------------------------------------------------
-
--- [CSL-1122] create a newtype for this instead?
-getAsciiString1b :: String -> Word8 -> Peek String
-getAsciiString1b typeName limit = getWord8 >>= \sz -> do
-            if sz > limit
-               then fail $ typeName ++ " shouldn't be more than "
-                                    ++ show limit ++ " bytes long"
-               else traverse checkAscii =<< BS.unpack <$> getBytes (fromIntegral sz)
-  where
-    checkAscii (chr . fromIntegral -> c) =
-        if isAscii c
-           then return c
-           else fail $ "Not an ascii symbol in " ++ typeName ++ " " ++ show c
-
-putAsciiString1b :: String -> Poke ()
-putAsciiString1b str =  putWord8 (fromIntegral $ length str)
-                     >> putBytes (BS.pack $ map (fromIntegral . ord) str)
-
-sizeAsciiString1b :: Size String
-sizeAsciiString1b = VarSize $ \s -> 1 + length s
-
--- | Get some bytes from the input.
-getBytes :: Int -> Peek ByteString
-getBytes i =
-    reifyNat (fromIntegral i) $
-        \(_ :: Proxy n) -> unStaticSize <$>
-        (Store.peek :: Peek (StaticSize n ByteString))
-
--- | Serialize some bytes.
---
--- (Unlike 'put' for 'ByteString', it doesn't write the number of bytes into
--- the stream, only the bytes themselves.)
-putBytes :: ByteString -> Poke ()
-putBytes bs =
-    reifyNat (fromIntegral $ BS.length bs) $ \(_ :: Proxy n) ->
-        Store.poke (Store.toStaticSizeEx bs :: StaticSize n ByteString)
+-- | Compute size of something serializable in bytes.
+biSize :: Bi a => a -> Byte
+biSize = fromIntegral . BS.length . serialize'
```

Comment:
This one is a bit of a temporary hack, and it has been discussed already in other places. Essentially, the old `Bi` typeclass included the `size` method to store (no pun intended!) the size (in bytes) of the serialised data structure. This is used in the [stripHashMap](https://github.com/input-output-hk/cardano-sl/blob/master/core/Pos/Util/Limits.hs#L40) function, and so I was forced to write this inefficient version which calculates it on the fly. In the future we should really look at how to resolve this in a better way.